apps/token/src/routes/governance/hooks/use-vote-information.ts

```diff
@@ -1,162 +1,185 @@
+  }, [proposal.votes.yes.totalEquityLikeShareWeight]);
+
+  const totalTokensVoted = useMemo(
     () => yesTokens.plus(noTokens),
     [yesTokens, noTokens]
   );
-  const yesPercentage = React.useMemo(
+
+  const totalELSWeight = useMemo(
+    () => yesELSWeight.plus(noELSWeight),
+    [yesELSWeight, noELSWeight]
+  );
+
+  const yesPercentage = useMemo(
     () =>
       totalTokensVoted.isZero()
         ? new BigNumber(0)
         : yesTokens.multipliedBy(100).dividedBy(totalTokensVoted),
     [totalTokensVoted, yesTokens]
   );
-  const noPercentage = React.useMemo(
+
+  const yesLPPercentage = useMemo(
+    () =>
+      totalELSWeight.isZero()
+        ? new BigNumber(0)
+        : yesELSWeight.multipliedBy(100).dividedBy(totalELSWeight),
+    [totalELSWeight, yesELSWeight]
+  );
+
+  const noPercentage = useMemo(
     () =>
       totalTokensVoted.isZero()
         ? new BigNumber(0)
         : noTokens.multipliedBy(100).dividedBy(totalTokensVoted),
     [noTokens, totalTokensVoted]
   );
-  const participationMet = React.useMemo(() => {
+
+  const noLPPercentage = useMemo(
+    () =>
+      totalELSWeight.isZero()
+        ? new BigNumber(0)
+        : noELSWeight.multipliedBy(100).dividedBy(totalELSWeight),
+    [noELSWeight, totalELSWeight]
+  );
+
+  const participationMet = useMemo(() => {
     const tokensNeeded = totalSupply.multipliedBy(requiredParticipation);
     return totalTokensVoted.isGreaterThan(tokensNeeded);
   }, [requiredParticipation, totalTokensVoted, totalSupply]);
 
-  const majorityMet = React.useMemo(() => {
+  const participationLPMet = useMemo(() => {
+    if (!requiredParticipationLP) {
+      return false;
+    }
+    const tokensNeeded = totalSupply.multipliedBy(requiredParticipationLP);
+    return totalELSWeight.isGreaterThan(tokensNeeded);
+  }, [requiredParticipationLP, totalELSWeight, totalSupply]);
+
+  const majorityMet = useMemo(() => {
     return yesPercentage.isGreaterThanOrEqualTo(requiredMajorityPercentage);
   }, [yesPercentage, requiredMajorityPercentage]);
 
-  const totalTokensPercentage = React.useMemo(() => {
+  const majorityLPMet = useMemo(() => {
+    return yesLPPercentage.isGreaterThanOrEqualTo(requiredMajorityLPPercentage);
+  }, [yesLPPercentage, requiredMajorityLPPercentage]);
+
+  const totalTokensPercentage = useMemo(() => {
     return totalTokensVoted.multipliedBy(100).dividedBy(totalSupply);
   }, [totalTokensVoted, totalSupply]);
 
-  const willPass = React.useMemo(
+  const totalLPTokensPercentage = useMemo(() => {
+    return totalELSWeight.multipliedBy(100).dividedBy(totalSupply);
+  }, [totalELSWeight, totalSupply]);
+
+  const willPassByTokenVote = useMemo(
     () =>
       participationMet &&
       new BigNumber(yesPercentage).isGreaterThanOrEqualTo(
         requiredMajorityPercentage
       ),
     [participationMet, requiredMajorityPercentage, yesPercentage]
   );
+
+  const willPassByLPVote = useMemo(
```

Comment:
suggestion (non blocking): I _really_ dislike this huge chain of useMemos. If possible lets combine this all into a few useMemos (or just one ?!) that read more linearly